### ISSUE TITLE
Android App Connection Issue

### ISSUE THREAD

**Issue by stollo770** [owncloud/android#885]
Hello,

I am having a problem with my android application in that it cannot sync back to my server from remote. I can connect to ownlcoud via the web interface from http://mydomain.ddns.net/owncloud just fine and my desktop application is working perfectly, but the android application is not. 

When ever I try to connect to owncloud via the android application it keeps saying synchronization failed with a longer messages that says synchronization of admin@mydomain.ddns.net/owncloud could not be completed. Its super frustrating because it CAN connect to my server and it DOES allow me to log in (albiet via an unsecure connection). When I log into my server using the app nothing is displayed and when I swipe down to refresh/sync I get the error message. Also my owncloud instance logs an error of: Login failed: ''( Remote IP: 'IP.FROM.THE.INTERNET', X-Forwarded-For: '')". What does this mean? Also its worth nothing that the Remote IP it is talking about is completely different from the IP I forward for my owncloud. Meaning if my remote IP was 66.235.232.435 (<fake btw) then the error would log a remote IP 78.454.654.456 (<fake also). 

Does anyone know what is going on? Does the app require a secure connection to reach owncloud like https or something? I really don't want to fiddle around with encryption because I'm not keeping ultra sensitive material on my owncloud, just various photos and files. I also know very little about this subject being that I only installed owncloud a week ago.

Any help would be much appreciated. Thank you.

**Comment by masensio** [comment]
Hi @stollo770,

thanks for your feedback and for deciding to use owncloud.

OwnCloud runs with https and also with http. But maybe there is something misconfigured in your server. 

Could you please use the template https://github.com/owncloud/android/blob/master/issue_template.md to describe your problem? 
The more information, the better we can help you.

**Comment by stollo770** [comment]
### Expected behaviour
I should be able to log in with my owncloud android client and/or my android webdav client

### Actual behaviour
The login fails every single time

### Steps to reproduce
1. Open owncloud on android / open webdav client
2. enter necessary login information mydoman.ddns.net/owncloud  |user: admin | |password: the password I use|
3. Failure

### Environment data
Android version:
4.4.2
Device model: 
Samsung SM G900A
Stock or customized system:
Nope, its what came with the phone, though it is rooted.
ownCloud app version:
1.6.2
ownCloud server version:
7.0.3
### Logs
#### Web server error log
```
From OwnCloud Application: Login failed: ''( Remote IP: 'IP.FROM.THE.INTERNET', X-Forwarded-For: '')"

From Webdav: "Sabre_DAV_Exception_NotAuthenticated No basic authentication headers were found"
```

#### ownCloud log (data/owncloud.log)
```
Not sure if these are correct, or relevant for my situation, but here: 

{"app":"PHP","message":"touch(): Unable to create file C:\/wamp\/www\/owncloud\/data\/owncloud.log because No such file or directory at C:\\wamp\\www\\owncloud\\lib\\private\\log\\owncloud.php#49","level":3,"time":"2015-02-13T03:37:35+00:00"}

{"app":"PHP","message":"filemtime(): stat failed for F:\\Owncloud\/admin\/54dd7211048d2 at C:\\wamp\\www\\owncloud\\lib\\private\\files\\storage\\mappedlocal.php#155","level":3,"time":"2015-02-13T03:40:01+00:00"}

{"app":"PHP","message":"filemtime(): stat failed for F:\\Owncloud\/admin\/54dd7211048d2 at C:\\wamp\\www\\owncloud\\lib\\private\\files\\storage\\mappedlocal.php#155","level":3,"time":"2015-02-13T03:40:02+00:00"}

Here is what the OwnCloud App says: synchronization of admin@myremotedomain.ddns.net could not be completed

```

Please help me, I'll take a solution that will work for either the owncloud app or a webdav client. I just have to have owncloud access on my phone. Half of why I even use cloud drives is because they are great to transfer data to and from phones. Also, my owncloud is hosted on a Windows 7 PC running the Wamp server. Please don't say I need linux, I don't want to use linux. Thank you for any and all help.

**Comment by guaka** [comment]
I think I'm experiencing the same with owncloud on ubuntu.

Owncloud OSX desktop can connect but the Android app is stuck already with "Couldn't find host". 
I have the suspicion that it somehow requires https, but I just want to try out owncloud locally with a server in my own home so I don't care too much about https for that.

**Comment by guaka** [comment]
I set up a self signed certificate on my box.

Then the android app actually asked whether to trust my certificate or not, after which I still ended up with 
"Couldn't find host".

Weird weird weird.

**Comment by guaka** [comment]
Actually. If I use https://192.168.1.x/ it says "Server instance not found".
Then with https://192.168.1.x/owncloud I briefly get "Testing connection" and then "Unknown error occurred!"

**Comment by guaka** [comment]
Weird enough I get almost exactly the same results with [
BezKloboukuNos' Client for ownCloud"](https://play.google.com/store/apps/details?id=com.owncloud.androidApp)

**Comment by tobiasKaminsky** [comment]
@guaka Have you tried it with the official ownclound app and the official owncloud server?
demo.owncloud.org
User: test 
Password: test

Please try it.

**Comment by guaka** [comment]
@tobiasKaminsky 

Meanwhile I found out the following, by opening the URL in my Android browser
- `host.local` or `host` doesn't work on my Android devices
- IP address works, but then my ownCloud server wanted it to be `host.local` unless I go to the admin interface and also allow access through the IP address

Ideally the Android app would pass through the error message coming from ownCloud instead of simply showing "Unknown error occurred".

I'm sorry for hijacking the original issue... although my actual resolution could be the same.

**Comment by tobiasKaminsky** [comment]
> Ideally the Android app would pass through the error message coming from ownCloud instead of simply showing "Unknown error occurred".
@davivel  this is a good idea. Maybe also show the proposal to try it with demo.owncloud.org to verify the functionality of the app? (see #893 for details, as we should focus on the acceptance of the app...)

**Comment by robertwenner** [comment]
Just in case this is helpful to someone: I could not connect to my owncloud server from the phone, but from my laptop it worked just fine. The owncloud app always said "couldn't find host" and sometimes "unknown error occurred". The culprit was my router, which had 8.8.8.8 as static DNS. In the WLAN advanced settings in android that was the first name server, and the router was second. Of course, Google DNS cannot resolve my internal names. Removing that in the router made connections to my owncloud work.

**Comment by Jef2i** [comment]
Hi,
Exact same issue as the one reported by @stollo770 .
But little different context:
* Android 4.4.4, Client : 1.6.2 (fallback from 1.7.0 to see if it was a version specific issue)
* OC Server : 7.0.3 on Synology NAS

Works like a sharm my computer inside my home network (linux client) so as from internet (Windows clients)
Error popup message on android : "synchronization of login@mydomain could not be completed"

Note that I also have enabled the mozilla_sync 3rdparty app and I have a similar issue, working from computers but not on android.
For that, on computers, It was necessary to downgrade to firefox before v29, then register, then upgraded.
Archived versions are also available for android, but the registering process fails

I ve generated a custom web site certificate too.
Imported into android's system wide environment (certificate management section )

_Log messages on android when trying to sync from OC app_ : 
"failed to sync operation logindomain u0 (owncloud), org.owncloud, USER, latestRunTime 147300280, reason 1000, SyncResult: tooManyRetries: true delayUnit: 86400 stats []"
followed by another log message
"not retrying sync operation because it retried too many times ... blabla"

_Log messages on android when trying to sync from Firefox Sync (summed up)_ :
"javax.net.ssl.SSLPeerUnverifiedException : No peer certificate
at com.android.org.conscrypt.SSL.SessionImpl.getPeerCertificate(...
"

What about an SSLayer java issue ?

**Comment by Jef2i** [comment]
Hi,
Just to report that I still can't use owncloud client on my android device since a while now.
And maybe due to this problem.

I was wondering if there were any clue to resolve it...
If I may help conserning it, please let us know.

Thanks in advance !

**Comment by tobiasKaminsky** [comment]
@Jef2i 
Have you tried it with:
demo.owncloud.org
User: test
Password: test

Maybe you can also test the connection to your server without https?

**Comment by Jef2i** [comment]
Well, reading your reply i've been a bit despointed.
Why should I test with the demo site ? Which new elements this test could give us ??
Then I realised that I didn't test http connexion to my own cloud.
What I did with the demo site ! :-) 
Results are that I'm facing with similar issues trying to sync with demo, securely or not.
Thoughts !
Confirmations that issues are coming from android side.
Then it remainded me that I'm still having installed the "JB workaround" installed along with the OC app. And I'm running 4.4.4 KK...
Removed everything, then installed only OC app, then registered securely to my own cloud.
Working finally !!

Since I didn't try to reinstall the "workaround" app to confirm (or maybe not) that it was the reason of my problem...
